go/libraries/doltcore/sqle/sqlddl_test.go

```diff
@@ -1186,6 +1186,110 @@ INSERT INTO child_non_unq VALUES ('1', 1), ('2', NULL), ('3', 3), ('4', 3), ('5'
 	}
 }
 
+func TestDropIndex(t *testing.T) {
+	ctx := context.Background()
+	dEnv := dtestutils.CreateTestEnv()
+	root, err := dEnv.WorkingRoot(ctx)
+	if err != nil {
+		panic(err)
+	}
+	// setup
+	root, err = ExecuteSql(dEnv, root, "create table parent (i int, j int, k int, index i (i), index ij (i, j), index ijk (i, j, k), index j (j), index kji (k, j, i));")
+	require.NoError(t, err)
+	root, err = ExecuteSql(dEnv, root, "create table child (x int, y int, constraint fk_child foreign key (x, y) references parent (i, j));")
+	require.NoError(t, err)
+
+	// check foreign keys for updated index
+	fkc, err := root.GetForeignKeyCollection(ctx)
+	require.NoError(t, err)
+	fkChild, ok := fkc.GetByNameCaseInsensitive("fk_child")
+	require.True(t, ok)
+	require.Equal(t, "xy", fkChild.TableIndex)
+	require.Equal(t, "ij", fkChild.ReferencedTableIndex)
+
+	// add primary key
+	root, err = ExecuteSql(dEnv, root, "alter table parent add primary key (i, j);")
+	require.NoError(t, err)
+
+	// dropping secondary index ij, should choose ijk
+	root, err = ExecuteSql(dEnv, root, "alter table parent drop index ij;")
+	require.NoError(t, err)
+
+	// check foreign keys for updated index
+	fkc, err = root.GetForeignKeyCollection(ctx)
+	require.NoError(t, err)
+	fkChild, ok = fkc.GetByNameCaseInsensitive("fk_child")
+	require.True(t, ok)
+	require.Equal(t, "xy", fkChild.TableIndex)
+	require.Equal(t, "ijk", fkChild.ReferencedTableIndex)
+
+	// dropping secondary index ijk, should switch to primary key
+	root, err = ExecuteSql(dEnv, root, "alter table parent drop index ijk;")
+	require.NoError(t, err)
+
+	// check foreign keys for updated index
+	fkc, err = root.GetForeignKeyCollection(ctx)
+	require.NoError(t, err)
+	fkChild, ok = fkc.GetByNameCaseInsensitive("fk_child")
+	require.True(t, ok)
+	require.Equal(t, "xy", fkChild.TableIndex)
+	require.Equal(t, "", fkChild.ReferencedTableIndex)
+
+	// no viable secondary indexes left, should be unable to drop primary key
+	_, err = ExecuteSql(dEnv, root, "alter table parent drop primary key;")
+	require.Error(t, err)
+
+	// drop tables
+	root, err = ExecuteSql(dEnv, root, "drop table child;")
+	require.NoError(t, err)
+	root, err = ExecuteSql(dEnv, root, "drop table parent;")
+	require.NoError(t, err)
+
+	// setup
+	root, err = ExecuteSql(dEnv, root, "create table parent (i int);")
```

Comment:
You have two separate test cases in this method. Name them and run them in t.Run() blocks